### PR TITLE
fix: cli list return nothing when list is empty

### PR DESCRIPTION
**Pull request by zhongwencool**
Fixes https://emqx.atlassian.net/browse/EMQX-10950 and https://emqx.atlassian.net/browse/EMQX-11004
Also add new command:
```
./bin/emqx ctl retainer topics <Start> <Length>
```

### <samp>🤖 Generated by Copilot at f4a88f7</samp>

This pull request improves the CLI commands for emqx management, observer, and retainer modules, by adding features, handling edge cases, and fixing bugs. It also updates the version numbers of the emqx_modules and emqx_retainer applications, and adds a test case for the retainer command.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/(ce|ee)/(feat|perf|fix)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

**Review by sstrigler — DISMISSED** [review]

File: apps/emqx_retainer/src/emqx_retainer_mnesia_cli.erl [hunk 1]
```diff
@@ -65,7 +66,9 @@ retainer(_) ->
     emqx_ctl:usage(
         [
             {"retainer info", "Show the count of retained messages"},
-            {"retainer topics", "Show all topics of retained messages"},
+            {"retainer topics", "Same as retainer topic 1 1000"},
+            {"retainer topics <Start> <Length>",
```
Comment (zmstone) [inline apps/emqx_retainer/src/emqx_retainer_mnesia_cli.erl:70]:
```suggestion
            {"retainer topics <Start> <Limit>",
```

**Review by qzhuyan — APPROVED** [review]

**Review by savonarola — APPROVED** [review]
